src/tmpfiles/tmpfiles.c

```diff
@@ -2633,6 +2750,51 @@ static int find_gid(const char *group, gid_t *ret_gid, Hashmap **cache) {
         return name_to_gid_offline(arg_root, group, ret_gid, cache);
 }
 
+static int parse_age_by_from_arg(const char *age_by_str, AgeBy *ret_age_by) {
+        _cleanup_free_ char *a = NULL;
+        const char *s = NULL;
+        AgeBy ab = 0;
+
+        static const struct {
+                char age_by_chr;
+                AgeBy age_by_flag;
+        } age_by_types[] = {
+                { 'a', AGE_BY_ATIME_F },
+                { 'b', AGE_BY_BTIME_F },
+                { 'c', AGE_BY_CTIME_F },
+                { 'm', AGE_BY_MTIME_F },
+                { 'A', AGE_BY_ATIME_D },
+                { 'B', AGE_BY_BTIME_D },
+                { 'M', AGE_BY_MTIME_D },
+        };
+
+        assert(age_by_str);
+        assert(ret_age_by);
+
+        s = age_by_str;
+
+        if (isempty(s))
+                return -EINVAL;
+
+        do {
+                /* Allow valid characters (whitespaces are ignored). */
+                if (!IN_SET(*s, 'a', 'b', 'c', 'm', 'A', 'B', 'M', ' ', '\t'))
+                        return -EINVAL;
+
+                for (unsigned int i = 0; i < ELEMENTSOF(age_by_types); i++)
```

Comment:
`i` could have been `size_t` (even though `unsigned` works)' will update in the next iteration.